### PR TITLE
source-mysql: upgrade debezium to 2.5.4

### PR DESCRIPTION
**Pull request by theyueli**
This patch attempts to fix the following issues:

https://github.com/airbytehq/oncall/issues/5332
https://github.com/airbytehq/airbyte/issues/32348

### CODE REVIEW

File: airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/cdc/MySqlDebeziumStateUtil.java [hunk 1]
```diff
@@ -63,9 +64,9 @@ public boolean savedOffsetStillPresentOnServer(final JdbcDatabase database, fina
         LOGGER.info("Connector used GTIDs previously, but MySQL server does not know of any GTIDs or they are not enabled");
         return false;
       }
-      final GtidSet gtidSetFromSavedState = new GtidSet(savedState.gtidSet().get());
+      final MySqlGtidSet gtidSetFromSavedState = new MySqlGtidSet(savedState.gtidSet().get());
```
Comment (xiaohansong) [inline airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/cdc/MySqlDebeziumStateUtil.java:67]: I think they also have an implementation for [MariaDbGtidSet](https://github.com/debezium/debezium/blob/main/debezium-connector-mariadb/src/main/java/io/debezium/connector/mariadb/gtid/MariaDbGtidSet.java)

Can you test if this change would work on a maria db?

Comment (theyueli) [reply airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/cdc/MySqlDebeziumStateUtil.java:67]: yes, from my previous experience, mariadb and original MySQL are almost considered two different databases now with different storage engines. We probably shall really have another connector called mariadb that uses the GtidSet you mentioned above?

**Review by xiaohansong — APPROVED** [review]
LGTM!

**Comment by theyueli** [comment]
a quick update @xiaohansong @evantahler regarding mariadb compatibility:

I installed the latest Mariadb on my laptop and ran a full and incremental sync against it using my development image from this PR. The sync works correctly.

File: airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/cdc/MySqlDebeziumStateUtil.java (hunk 1 above)
Comment (theyueli) [reply airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/cdc/MySqlDebeziumStateUtil.java:67]: Given the latest test on mariadb, I think we can probably continue using MySQL connector for mariadb ...

**Pull request merged**